lib/compress.js

```diff
@@ -2130,9 +2130,6 @@ merge(Compressor.prototype, {
                     extract_declarations_from_unreachable_code(compressor, self.body, a);
                     return make_node(AST_BlockStatement, self, { body: a });
                 }
-            } else {
-                // self instanceof AST_Do
-                return self.body;
```

Comment:
I prefer being more explicit and keeping this `else` and have it `return self;`. At the very least it avoids the `if` below.